src/components/AvatarStack/AvatarStack.tsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+
+import {block} from '../utils/cn';
+import {isOfType} from '../utils/isOfType';
+
+import {AvatarStackItem} from './AvatarStackItem';
+import {AvatarStackMoreButton} from './AvatarStackMoreButton';
+import i18n from './i18n';
+import type {AvatarStackProps} from './types';
+
+import './AvatarStack.scss';
+
+const b = block('avatar-stack');
+
+const isMoreButton = isOfType(AvatarStackMoreButton);
+
+const AvatarStackComponent = ({
+    max = 3,
+    overlapSize = 's',
+    children,
+    className,
+}: AvatarStackProps) => {
+    const moreButton: React.ReactElement[] = [];
+    const visibleItems: React.ReactElement[] = [];
+    let moreItems = 0;
+
+    React.Children.forEach(children, (child) => {
+        if (!React.isValidElement(child)) {
+            return;
+        }
+
+        const isButton = isMoreButton(child);
+        const item = (
+            <AvatarStackItem
+                key={isButton ? `more-button-${moreButton.length}` : visibleItems.length}
+            >
+                {child}
+            </AvatarStackItem>
+        );
+
+        if (isButton) {
+            moreButton.push(item);
+        } else if (visibleItems.length <= max) {
+            visibleItems.unshift(item);
+        } else {
+            moreItems += 1;
+        }
+    });
+
+    const hasMoreButton = moreItems > 0;
+    /** Avatars + more button, or just avatars, when avatars count is equal to `max` or less */
+    const normalOverflow = moreItems >= 1;
+
+    if (!moreButton.length && hasMoreButton) {
+        const guessedSize = visibleItems[0]?.props.children.props.size;
+        moreButton.push(
+            <AvatarStackItem key="more-button">
+                <AvatarStackMoreButton
+                    size={guessedSize}
+                    count={moreItems + 1}
+                    aria-label={i18n('more')}
```

Comment:
Why we provide `onClick` and not use it? Default MoreButton does nothing when clicked

Reply:
This component for user usage primary, click handler could be required